### PR TITLE
per gke's suggestion, use dash instead of under.

### PR DESCRIPTION
**Pull request by andysim3d**
Also amend sample yaml file, move selector label to metadata label.

### CODE REVIEW

File: cmd/operator/deploy/operator/05-deployment.yaml [hunk 1]
```diff
@@ -18,14 +18,14 @@ metadata:
   namespace: gmp-system
   name: gmp-operator
   labels:
+    app: managed_prometheus_operator
```
Comment (pintohutch) [inline cmd/operator/deploy/operator/05-deployment.yaml:21]: It looks like you moved this without reformatting to dashes.

**Review by pintohutch — CHANGES_REQUESTED** [review]
Also, be sure to run `make presubmit` to regenerate affected manifests.

File: cmd/operator/deploy/operator/05-deployment.yaml (hunk 1 above)
Comment (andysim3d) [reply cmd/operator/deploy/operator/05-deployment.yaml]: done

File: cmd/operator/deploy/operator/05-deployment.yaml [hunk 2]
```diff
@@ -18,21 +18,21 @@ metadata:
...
   name: gmp-operator
   labels:
+    app: managed-prometheus-operator
     app.kubernetes.io/name: gmp-operator
     app.kubernetes.io/component: operator
     app.kubernetes.io/part-of: gmp
 spec:
   replicas: 1
   selector:
     matchLabels:
```
Comment (pintohutch) [inline cmd/operator/deploy/operator/05-deployment.yaml:28]: Can you add it back here and rerun `make presubmit`? It should regenerate the latest `examples/operator.yaml` with your change here.

Comment (andysim3d) [reply cmd/operator/deploy/operator/05-deployment.yaml:28]: done.

**Review by pintohutch — APPROVED** [review]

**Pull request merged**